src/MOI_wrapper.jl

```diff
@@ -1479,31 +1512,57 @@ function _store_solution(model::Optimizer, ret::Cint)
     resize!(x.rowvalue, numRows)
     resize!(x.rowdual, numRows)
     resize!(x.rowstatus, numRows)
-    # Load the solution if optimal.
-    if Highs_getModelStatus(model) == Cint(kOptimal)
-        Highs_getSolution(model, x.colvalue, x.coldual, x.rowvalue, x.rowdual)
-        Highs_getBasis(model, x.colstatus, x.rowstatus)
-        x.has_solution = true
-        return
+    status = Highs_getModelStatus(model)
+    statusP = Ref{Cint}()
+    if status == Cint(kInfeasible)
+        ret = Highs_getDualRay(model, statusP, x.rowdual)
+        _check_ret(ret)
+        x.has_dual_ray = statusP[] == 1
+    elseif status == Cint(kUnbounded)
+        ret = Highs_getPrimalRay(model, statusP, x.colvalue)
+        _check_ret(ret)
+        x.has_primal_ray = statusP[] == 1
+    else
+        Highs_getIntInfoValue(model, "primal_solution_status", statusP)
+        x.has_primal_solution = statusP[] == 2
+        Highs_getIntInfoValue(model, "dual_solution_status", statusP)
+        x.has_dual_solution = statusP[] == 2
+        if x.has_primal_solution
+            Highs_getSolution(
+                model,
+                x.colvalue,
+                x.coldual,
+                x.rowvalue,
+                x.rowdual,
+            )
+            Highs_getBasis(model, x.colstatus, x.rowstatus)
+        end
     end
-    # Check for a certificate of primal infeasibility.
-    has_ray = Ref{Cint}(0)
-    ret = Highs_getDualRay(model, has_ray, x.rowdual)
-    x.has_dual_ray = has_ray[] == 1
-    _check_ret(ret)
-    if x.has_dual_ray
-        return
+    return
+end
+
+function _enforce_binary_bounds(f, model::Optimizer)
```

Comment:
what's the `f` side-effecting function here? Might be good to have a comment, otherwise it's cryptic from reading the code

Reply:
(even if internal)

Reply:
Yeah I may just refactor this back into `MOI.optimize!`. I thought it was going to be more complicated.